src/cdk-experimental/listbox/listbox.ts

```diff
@@ -8,182 +8,146 @@
   AfterContentInit,
+  ChangeDetectorRef,
   ContentChildren,
   Directive,
   ElementRef,
-  EventEmitter,
   forwardRef,
   Inject,
   Input,
   OnDestroy,
-  OnInit,
   Optional,
   Output,
   QueryList,
 } from '@angular/core';
 import {ActiveDescendantKeyManager, Highlightable, ListKeyManagerOption} from '@angular/cdk/a11y';
 import {DOWN_ARROW, ENTER, LEFT_ARROW, RIGHT_ARROW, SPACE, UP_ARROW} from '@angular/cdk/keycodes';
 import {BooleanInput, coerceArray, coerceBooleanProperty} from '@angular/cdk/coercion';
-import {SelectionChange, SelectionModel} from '@angular/cdk/collections';
-import {defer, merge, Observable, Subject} from 'rxjs';
-import {startWith, switchMap, takeUntil} from 'rxjs/operators';
+import {SelectionModel} from '@angular/cdk/collections';
+import {BehaviorSubject, defer, merge, Observable, Subject} from 'rxjs';
+import {filter, mapTo, startWith, switchMap, takeUntil} from 'rxjs/operators';
 import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
 import {Directionality} from '@angular/cdk/bidi';
 import {CDK_COMBOBOX, CdkCombobox} from '@angular/cdk-experimental/combobox';
 
-let nextId = 0;
-let listboxId = 0;
+/** The next id to use for the CdkOption directive. */
+let nextOptionId = 0;
 
-export const CDK_LISTBOX_VALUE_ACCESSOR = {
-  provide: NG_VALUE_ACCESSOR,
-  useExisting: forwardRef(() => CdkListbox),
-  multi: true,
-};
+/** The next id to use for the CdkListbox directive. */
+let nextListboxId = 0;
 
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     'role': 'option',
     'class': 'cdk-option',
-    '(click)': 'toggle()',
-    '(focus)': 'activate()',
-    '(blur)': 'deactivate()',
     '[id]': 'id',
-    '[attr.aria-selected]': 'selected || null',
-    '[attr.tabindex]': '_getTabIndex()',
-    '[attr.aria-disabled]': '_isInteractionDisabled()',
-    '[class.cdk-option-disabled]': '_isInteractionDisabled()',
-    '[class.cdk-option-active]': '_active',
-    '[class.cdk-option-selected]': 'selected',
+    '[attr.aria-selected]': 'isSelected() || null',
+    '[attr.tabindex]': '!listbox.useActiveDescendant && !disabled ? -1 : null',
+    '[attr.aria-disabled]': 'disabled',
+    '[class.cdk-option-disabled]': 'disabled',
+    '[class.cdk-option-active]': 'isActive()',
+    '[class.cdk-option-selected]': 'isSelected()',
+    '(click)': '_clicked.next()',
   },
 })
-export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable {
-  private _selected: boolean = false;
-  private _disabled: boolean = false;
-  private _value: T;
-  _active: boolean = false;
+export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightable, OnDestroy {
+  /** The id of the option's host element. */
+  @Input() id = `cdk-option-${nextOptionId++}`;
 
-  /** The id of the option, set to a uniqueid if the user does not provide one. */
-  @Input() id = `cdk-option-${nextId++}`;
+  // TODO(mmalerba): What do we do if the user doesn't specify a value?
```

Comment:
It's kind of annoying for users when they have to do:

```
<div cdkOption cdkOptionValue="blue">Blue</div>
```

If I just set the default value to `null` the listbox feels broken because clicking on one option selects all of them. Some ideas I had to make this better are:
- Track a `SelectionModel<CdkOption>` instead of `SelectionModel<T>` and allow the selection to contain duplicate values. e.g. if you checked 3 options with the default value, the listbox's value would be `[null, null, null]`
- Bind to the directive host selector: `<div cdkOption="blue">Blue</div>`
- Use the `innerText` or `textContent` as the default value: `<div cdkOption>Blue</div>` (value = 'Blue')

Reply:
My vote is either for binding to the directive host selector or throwing an error if a value isn't assigned by the time `ngAfterViewInit` fires.

Tracking options has been annoying in `mat-select`, because you also have to add/remove options as the list changes which can lead to "changed after checked" errors.

Using the `innerText` will be annoying as well, because we'll have to monitor the DOM for changes.

Reply:
+1 to making it `<li cdkOption="blue">`

While I wish it was easier to fall back to the text content, I think it's reasonable to error when no value is specified

Reply:
SGTM, but note we can't throw an error for `<li cdkOption>` because I don't think we can really distinguish it from `<li cdkOption="">`. My worry about this is that people will do something like:

```
<ul cdkListbox>
  <li cdkOption>apple</li>
  <li cdkOption>orange</li>
</ul>
```

And then be confused when clicking either of the options selects all of the options. I can think of one thing we can do to mitigate this confusion which is maybe add a dev mode warning when 2 options have the same value and refer people to the docs section that talks about it. WDYT?

Reply:
Yeah, I think that's fine